### PR TITLE
MLT readout map rework

### PR DESCRIPTION
**Pull request by MRiganSUSX**
Fix for custom readout map used in MLT to only apply custom readout windows to explicitly configured TC types.

### CODE REVIEW

**Comment by MRiganSUSX** [comment]
More details - with description, configurations, testing, and results - available here: 
[custom_readout_map_fix.pdf](https://github.com/DUNE-DAQ/trigger/files/13377208/custom_readout_map_fix.pdf)
Required: [daqconf PR412](https://github.com/DUNE-DAQ/daqconf/pull/412)

File: plugins/ModuleLevelTrigger.cpp [hunk 1]
```diff
@@ -509,12 +509,12 @@ ModuleLevelTrigger::add_tc(const triggeralgs::TriggerCandidate& tc)
 
     for (std::vector<PendingTD>::iterator it = m_pending_tds.begin(); it != m_pending_tds.end();) {
       if (check_overlap(tc, *it)) {
-        if (m_use_readout_map) {
+	it->contributing_tcs.push_back(tc);
+        if ( (m_use_readout_map) && (m_readout_window_map.count(tc.type)) ){
```
Comment (ArturSztuc) [inline plugins/ModuleLevelTrigger.cpp:513]: I cannot select this all, but lines 511 - 536, that's a very, nested if statement,  makes it difficult to read and understand... normally something like this:

```
for(i = 0; i < n; ++i){
   if(this_true){
      if(that_als_true){
         50 lines of code
      }
   }
}
```

can be easily changed withs simple guard clauses:

```
for(i = 0; i < n; ++i){
   if(!this_true)
      continue;
   if(!that_also_true)
      continue;

   50 lines of code
}
```

Can read more about them on the refactoring guru page: https://refactoring.guru/replace-nested-conditional-with-guard-clauses

It's a comment though, PR still approved :+1:

**Review by ArturSztuc — APPROVED** [review]
LGME, although future guard clauses in MLT would be nice for readability.

**Review by bieryAtFnal — APPROVED** [review]
LGTM.

I've run tests in which I've separately controlled the readout window for two individual TC types in an emulated-data system in which I have three types of TCs being produced (timing, prescale, random).

As part of this, I made some enhancements to the `HDF5LIBS_TestDumpRecord` test application to print out the request window that was used for each SourceID in a TriggerRecord on disk.  I'll contribute those independently of this PR.

**Pull request merged**